src/EnhancedEcommerceService.js

```diff
@@ -104,10 +104,17 @@ class EnhancedEcommerceService {
         
         if(defaultEcommerceSchema){
             // The items array can include up to 200 elements.
-            data.length = Math.min(data.length, 200);
+            if (data != null) {
+                data.length = Math.min(data.length, 200);
+            }
             eventData.ecommerce = {};
-            eventData.ecommerce["items"] = data;
+            eventData.ecommerce["items"] = data | [];
```

Comment:
I think this line should be this instead:
```js
eventData.ecommerce["items"] = data || [];
```
So two pipes (`||`) instead of a single pipe. A single pipe is a [bitwise OR operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_OR). While this would probably still work, it's not the correct way to do it.

You might also want to look at the [nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). It's similar to `||`, but only looks at `null` and `undefined` instead of `null`, `undefined`, 0, empty string, and `NaN`. Basically, the same as the `??` operator that C# has.